tests/unit/Controller/UserApiControllerTest.php

```diff
@@ -0,0 +1,115 @@
+<?php
+
+namespace OCA\Text\Controller;
+
+use OCA\Text\Service\SessionService;
+use OCP\Collaboration\Collaborators\ISearch;
+use OCP\AppFramework\Http\DataResponse;
+use \OCP\IRequest;
+use OCP\IUserManager;
+use OCP\IUserSession;
+use OCA\Text\Db\Session;
+use Test\TestCase;
+
+class UserApiControllerTest extends TestCase {
+	/**
+	 * @var \PHPUnit\Framework\MockObject\MockObject
+	 */
+	private $collaboratorSearch;
+	/**
+	 * @var \PHPUnit\Framework\MockObject\MockObject
+	 */
+	private $userSession;
+	/**
+	 * @var \PHPUnit\Framework\MockObject\MockObject
+	 */
+	private $userManager;
+	/**
+	 * @var \PHPUnit\Framework\MockObject\MockObject
+	 */
+	private $sessionService;
+	/**
+	 * @var UserApiController
+	 */
+	private $userApiController;
+	/**
+	 * @var IRequest|\PHPUnit\Framework\MockObject\MockObject
+	 */
+	private $request;
+
+	protected function setUp(): void {
+		parent::setUp();
+		$this->request = $this->createMock(IRequest::class);
+		$this->collaboratorSearch = $this->createMock(ISearch::class);
+		$this->userSession = $this->createMock(IUserSession::class);
+		$this->userManager = $this->createMock(IUserManager::class);
+		$this->sessionService = $this->createMock(SessionService::class);
+		$this->userApiController = new UserApiController(
+			'text',
+			$this->request,
+			$this->sessionService,
+			$this->collaboratorSearch,
+			$this->userManager,
+			$this->userSession
+		);
+	}
+
+	/**
+	 * @dataProvider dataTestUsersIndex
+	 */
+	public function testUsersIndex(int $documentId, int $sessionId, string $sessionToken, string $filter)
+	{
+		$session = new Session();
+		$session->setUserId('admin');
+		$this->sessionService
+			->expects($this->once())
+			->method('isValidSession')->willReturn(true);
+		$this->sessionService
+			->expects($this->once())
+			->method('getAllSessions')->willReturn([[
+				'userId' => 'admin',
+				'displayName' => 'admin',
+			]]);
+		$this->sessionService
+			->expects($this->once())
+			->method('getSession')->willReturn($session);
+		$this->collaboratorSearch
+			->expects($this->once())
+			->method('search')->willReturn([
+				[
+					'exact' => [
+						'users' => []
+					],
+					'users' => [
+						[
+							'label' => 'admin',
+							'subline' => '',
+							'icon' => 'icon-user',
+							'value' => [
+								'shareType' => 0,
+								'shareWith' => 'admin'
+							],
+							'shareWithDisplayNameUnique' => 'admin',
+							'status' => []
+						],
+					]
+				]
+			]);
+
+			$users = $this->userApiController->index(
+				$documentId,
+				$sessionId,
+				$sessionToken,
+				$filter
+			);
+			$this->assertInstanceOf(DataResponse::class, $users);
```

Comment:
Well, this test only tests if `$users` is a `DataResponse` object, right? It seems to miss the actual tests on the content of the response? :thinking:

Reply:
Fixed myself now :)